substrate/frame/support/src/storage/storage_noop_guard.rs

```diff
@@ -71,7 +71,8 @@ impl<'a> StorageNoopGuard<'a> {
 impl<'a> Drop for StorageNoopGuard<'a> {
 	fn drop(&mut self) {
 		// No need to double panic, eg. inside a test assertion failure.
-		if sp_std::thread::panicking() {
+		#[cfg(feature = "std")]
```

Comment:
If not add this, it will cause compilation with `--all-features` fail

Reply:
I think its fine for two reasons: `StorageNoopGuard` is only used for testing (std enabled) and if it were used in the runtime, then we just abort on panic anyway instead of unwinding.

Reply:
The change is fine. In `no_std` we are not supporting unwinding any way and the program exits directly. Thus, we will never end here.